moto/core/responses_custom_registry.py

```diff
@@ -41,8 +43,16 @@ def find(self, request: Any) -> Tuple[Optional[responses.BaseResponse], List[str
         )
         found = []
         match_failed_reasons = []
+
+        # Handle non-standard AWS endpoint hostnames from ISO regions or custom S3 endpoints.
+        url_with_standard_aws_domain = urlunparse(
+            get_equivalent_url_in_aws_domain(request.url)
+        )
+        request_with_standard_aws_domain = request.copy()
```

Comment:
Copying and preparing the URL is only necessary in some cases, but it currently happens for every incoming HTTP request. The possible performance downsides of that worry me a little.

Can we make the copy+prepare conditional, and only execute that part if the standardized URL is different from the original?

Reply:
Yeah that's a good call.  There's no guarantee that URLs will roundtrip unchanged through urlparse/urlunparse, so I put the modification check a little further up.  That will also save some string copying.